src/app/admin/packingSlotsTable/PackingSlotActions.ts

```diff
@@ -0,0 +1,162 @@
+import supabase from "@/supabaseClient";
+import { DatabaseError } from "@/app/errorClasses";
+import { GridRowId } from "@mui/x-data-grid";
+import { PackingSlotRow } from "@/app/admin/packingSlotsTable/PackingSlotsTable";
+
+export interface DBPackingSlotData {
+    primary_key?: string;
+    name: string;
+    is_shown: boolean;
+    order: number;
+}
+
+export interface DataGridPackingSlotRow {
+    id: string;
+    name: string;
+    is_shown: boolean;
+    order: number;
+}
+interface packingSlotTableRowData extends DataGridPackingSlotRow {
+    isNew: boolean;
+}
+
+export const fetchPackingSlots = async (): Promise<PackingSlotRow[]> => {
+    const { data, error } = await supabase.from("packing_slots").select().order("order");
+    if (error) {
+        throw new DatabaseError("fetch", "packing slots");
+    }
+
+    const processedData = data.map((row): PackingSlotRow => {
+        return {
+            name: row.name,
+            id: row.primary_key,
+            is_shown: row.is_shown,
+            order: row.order,
+            isNew: false,
+        };
+    });
+
+    return processedData;
+};
+
+const processRowData = (tableData: packingSlotTableRowData): DBPackingSlotData => {
+    const data = { ...tableData };
+    const newData: DBPackingSlotData = {
+        primary_key: data.id,
+        name: data.name,
+        is_shown: data.is_shown,
+        order: data.order,
+    };
+    return newData;
+};
+
+const processNewRowData = (tableData: packingSlotTableRowData): DBPackingSlotData => {
+    const data = { ...tableData };
+    const newData: DBPackingSlotData = {
+        name: data.name,
+        is_shown: data.is_shown,
+        order: data.order,
+    };
+    return newData;
+};
+
+export const createPackingSlot = async (tableData: packingSlotTableRowData): Promise<void> => {
+    const data = processNewRowData(tableData);
+    const { error } = await supabase.from("packing_slots").insert(data);
+
+    if (error) {
+        console.log(error);
```

Comment:
We should remove this `console.log`. There will be a ticket soon to start logging errors / warnings / information but in a more structured way.